### ISSUE TITLE
[BUG] Incorrect types for ESM package exports

### ISSUE THREAD

**Issue by jaydenseric** [microsoft/playwright#11955]
**Context:**
- Playwright Version: 1.18.1
- Operating System: macOS
- Node.js version: 17.4.0
- Browser: All
- Extra: `typescript@4.6.0-dev.20220207`

**Code Snippet**

With this `jsconfig.json`:

```json
{
  "compilerOptions": {
    "module": "node12",
    "noEmit": true,
    "strict": true
  },
  "typeAcquisition": {
    "enable": false
  }
}
```

And this package `devDependency`:

```json
{
  "typescript": "^4.6.0-dev.20220207"
}
```

And this package `scripts`:

```json
{
  "types": "tsc -p jsconfig.json",
}
```

In `a.mjs`:

```js
// @ts-check

/** @typedef {import("playwright").Page} Page */
```

In `b.mts`:

```ts
import type { Page } from "playwright";
```

If you run:

```
npm run types
```

You will see `tsc` can't resolve the types:

<img width="979" alt="Screen Shot 2022-02-09 at 11 32 50 am" src="https://user-images.githubusercontent.com/1754873/153099223-095d0f15-76dc-458c-ac90-de184c24dbd5.png">

**Describe the bug**

Playwright packages don't provide types correctly for when their ESM package exports are imported.

First discussed here:

https://github.com/microsoft/TypeScript/issues/47792#issuecomment-1032096501

When working on fixing the types, be sure to validate the fixes using the `tsc` CLI and not the VS Code editor issues UI as it currently incorrectly resolves the types, hiding problems (see https://github.com/microsoft/TypeScript/issues/47795).

This bug with Playwright types is currently blocking me from completing and publishing work in an OSS project; I haven't been able to figure out any sort of temporary manual workaround that works.

**Comment by pavelfeldman** [comment]
Thanks for the report! Is there a chance you could provide us with a repo that reproduces the problem? I'll set this to p3 which means we are collecting the feedback. Once it gets enough votes, provided there is a repro, we'll act on it!

**Comment by jaydenseric** [comment]
> Is there a chance you could provide us with a repo that reproduces the problem?

No thanks. I already gave Microsoft, a 2.299 trillion dollar company, significant free QA labour for two Microsoft projects (TypeScript and Playwright) and a detailed bug report. I'm not keen to maintain a repo just for this.

> provided there is a repro

I provided exact step by step reproduction steps above above in this issue, it's only 4 files and ~15 lines of code.

> Once it gets enough votes, …, we'll act on it!

Or you could decide to fix the reproducible bug now _before_ it affects millions of other people once they start using ESM with TypeScript, and unblock the open source work I'm doing now.

Microsoft already has a TypeScript issue (https://github.com/microsoft/TypeScript/issues/47792) to go over popular packages and fix the types for ESM, yet you don't seem keen to follow through with that here; is one hand not talking with the other?

You guys are being paid to work on this stuff. I'm altruistically putting all my love and energy into adding type safety to the open source ESM packages I maintain (now that it's possible with TypeScript v4.6.0-dev.x), I've been doing this full time without pay for months now, so please factor this in when considering my frustrated tone.

**Comment by midgleyc** [comment]
Filed https://github.com/midgleyc/playwright-11955 as a repro, just copying the instructions from the initial issue (which were solid). This isn't a promise to maintain it forevermore.

I think "enough votes" refers to +1 reactions on the main issue, so the +3 reactions on the follow-up comment are wasted, unless the aim was just to express discontent with Microsoft prioritising the issues they pick up.

**Comment by mxschmitt** [comment]
I was able to verify that its fixed on the tip-of-tree (`1.23.0-alpha-jun-7-2022`) which gets released as 1.23 in 1-2 weeks.
